src/base/searchengine.h

```diff
@@ -132,6 +137,7 @@ private slots:
     bool m_searchStopped;
     QTimer *m_searchTimeout;
     QByteArray m_searchResultLineTruncated;
+    QList<QProcess*> m_downloaders;
```

Comment:
>which is likely a subject for a separate PR

That's it. And other your suggestions too.
I just returned the old code with minimal changes to follow the current design. I'm not going to improve that.